ci/cirrus_general_ci.yml

```diff
@@ -3,6 +3,21 @@
 # cibuildwheel also runs tests during the wheel build process, so there's no need
 # to have duplication.
 
+modified_clone: &MODIFIED_CLONE
+  # makes sure that for a PR the CI runs against a merged main
+  clone_script: |
+    if [ -z "$CIRRUS_PR" ]; then
```

Comment:
Do we need this part at all? That should work unchanged, so I guess it was due to limited flexibility in yaml (can't do `if not PR: clone_script: ...`)?

Reply:
Defining a clone_script overrides the clone that the CI does. I don't know if your kind of alternative works. I'm not smart enough to figure that out. I was surprised that cirrus didn't merge the PR onto main as default behaviour. I'll have to ask why that is. They're pretty responsive at answering questions.